test/rules/assertions/checkParamNames.js

```diff
@@ -925,6 +925,18 @@ export default {
           }
       `,
     },
+    {
+      code: `
+          /**
+           * @param foo
+           * @param foo.a
+           * @param foo.b
+           */
+          function quux ({"a": A, b}) {
+
+          }
+      `,
+    },
```

Comment:
This looks good.  But could you add an example with a hyphen? I think that will make it more clear when the tests get added to the README (and you can do that in this PR by running `npm run create-readme` after you finish updating tests, btw) why one might wish to use a literal. Also, if numbers would work here, it'd be nice I think to have such an example as well.